### ISSUE TITLE
io.js fails on copying extracted folder

### ISSUE THREAD

**Issue by nicks** [Medium/phantomjs#275]
Reported by @corbinu in this comment:
https://github.com/Medium/phantomjs/issues/271#issuecomment-70585090

```'Copying extracted folder /var/folders/pt/_mpbrdv90p3741h3xyt253pw0000gn/T/phantomjs/phantomjs-1.9.8-macosx.zip-extract-1421713554555/phantomjs-1.9.8-macosx -> /Users/corbinu/Developer/tests/node_modules/phantomjs/lib/phantom'```

@corbinu can you post more complete repro steps, most notably (1) the complete log, and (2) the version of io.js that you're using?

**Comment by corbinu** [comment]
There is no log as it never errors out and every version after 1.0.1

**Comment by zaggino** [comment]
Same on Windows with `io.js 1.0.3`
```
npm info build C:\DEV\weather-forecast2\node_modules\karma-phantomjs-launcher\node_modules\phantomjs
npm info linkStuff phantomjs@1.9.13
npm info install phantomjs@1.9.13

> phantomjs@1.9.13 install C:\DEV\weather-forecast2\node_modules\karma-phantomjs-launcher\node_modules\phantomjs
> node install.js

Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-windows.zip
Saving to C:\Users\martinz\AppData\Local\Temp\phantomjs\phantomjs-1.9.8-windows.zip
Receiving...
  [========================================] 100% 0.0s
Received 7292K total.
Extracting zip contents
Removing C:\DEV\weather-forecast2\node_modules\karma-phantomjs-launcher\node_modules\phantomjs\lib\phantom
Copying extracted folder C:\Users\martinz\AppData\Local\Temp\phantomjs\phantomjs-1.9.8-windows.zip-extract-1421816704172\phantomjs-1.9.8-windows -> C:\DEV\weather-forecast2\node_modules\karma-phantomjs-launcher\node_modules\phantomjs\lib\phantom
```
and stays frozen forever...

**Comment by nicks** [comment]
I filed a bug against ncp for this https://github.com/AvianFlu/ncp/issues/79

**Comment by jprichardson** [comment]
I have imported `ncp` into `fs-extra` https://www.npmjs.com/package/fs-extra and now works on `io.js`. So maybe this would be an option. You could then also get rid of `mkdirp` and `rimraf` as dependencies as `fs-extra` uses `rimraf` and imported `mkdirp`.

**Comment by nicks** [comment]
merged @corbinu 's PR and released an npm package as 1.9.14

**Comment by nicks** [comment]
The PR has been rolled back, because we were getting reports of it breaking things. see https://github.com/Medium/phantomjs/issues/279 for more information.

It appears that fs-extra has its own set of problems. The fs.move claims to complete successfully, but the fs.chmod claims the moved file doesn't exist. fs-extra adds a lot of hooks into both these methods, and it is unclear which one is lying. I didn't spend a lot of time looking into it.

**Comment by corbinu** [comment]
Is this project on Travis CI I see the yml file but not seeing it on travis obviously want to get to the bottom of this

**Comment by nicks** [comment]
yep, looking at the travis ci build right now

**Comment by nicks** [comment]
OK, here's a travis run of the fs-extra changes: https://travis-ci.org/Medium/phantomjs/jobs/48503467
and it appears to work ok there...

**Comment by corbinu** [comment]
no thats your rollback... I am going to add my fork with the changes to travis also

**Comment by nicks** [comment]
an alternative solution might be to just fork ncp, because they do not seem interested in fixing their package.

**Comment by corbinu** [comment]
True that is your call

**Comment by corbinu** [comment]
Are you still supporting Node 0.6 and 0.8 given they are in the .travis?

**Comment by nicks** [comment]
travis ci has been in this weird state lately where you can specify node 0.6 and 0.8, but the version of npm they have installed by default is incompatible with 0.8 due to the semver changes. There's some incantation to tell it to use the right npm version, but for now, I just removed them.

**Comment by corbinu** [comment]
Builds fine for me to :/ weird https://travis-ci.org/corbinu/phantomjs

**Comment by corbinu** [comment]
I also tried installing v0.10.33 on my Ubuntu 14.04 server and then installing. I can't replicate the issue

**Comment by jprichardson** [comment]
> It appears that fs-extra has its own set of problems. The fs.move claims to complete successfully, but the fs.chmod claims the moved file doesn't exist. fs-extra adds a lot of hooks into both these methods, and it is unclear which one is lying. I didn't spend a lot of time looking into it.

If anyone can help me reproduce the problems, that'd be incredibly helpful. `fs-extra` doesn't hook any methods.

**Comment by nicks** [comment]
I'm able to repro, and am trying to narrow it down to a nice repro case. I'm pretty sure it's related to fs.move across devices.

**Comment by nicks** [comment]
filed as https://github.com/jprichardson/node-fs-extra/issues/108

**Comment by jprichardson** [comment]
I fixed https://github.com/jprichardson/node-fs-extra/issues/108 and published `fs-extra@0.16.0`. You can try it. If for some reason `fs.move` still didn't work, you could stick with `fs.copy` like what you had previously (https://github.com/Medium/phantomjs/commit/8a96d6048541372a09d3cc5d2fb5f95146383a14) specifically keeping lines: https://github.com/Medium/phantomjs/blob/master/install.js#L351-L360. Then at least you'd have `io.js` support immediately.

**Comment by nicks** [comment]
ok, released as v1.9.15
